frontends/common/constantFolding.cpp

```diff
@@ -87,6 +89,9 @@ const IR::Expression *DoConstantFolding::getConstant(const IR::Expression *expr)
 }
 
 const IR::Node *DoConstantFolding::postorder(IR::PathExpression *e) {
+    if (ConstantFolding::filter_hook) {
+        if (auto *rv = ConstantFolding::filter_hook(this, e)) return rv;
```

Comment:
I don't like at all that the filter hook is taking `IR::Expression`, but hooking only to the `IR::PathExpression` case.

Reply:
Yes, I was of mixed mind about this.  Hooking everything would be more expensive, having every method call the hook and most of the time returning nullptr.  The compromise of only calling for PathExpression should probably also change to having the hook function take a PathExpression rather than an Expression.